JBBCode/Tokenizer.php

```diff
@@ -23,26 +23,20 @@ class Tokenizer
      */
     public function __construct($str)
     {
-        $strStart = 0;
         $strLen = strlen($str);
-        for ($index = 0; $index < $strLen; ++$index) {
-            if (']' == $str[$index] || '[' == $str[$index]) {
-                /* Are there characters in the buffer from a previous string? */
-                if ($strStart < $index) {
-                    $this->tokens[] = substr($str, $strStart, $index - $strStart);
-                    $strStart = $index;
-                }
+        $position = 0;
 
-                /* Add the [ or ] to the tokens array. */
-                $this->tokens[] = $str[$index];
-                $strStart = $index+1;
+        while($position < $strLen) {
+            $offset = strcspn($str, '[]', $position);
+            //Have we hit a single ']' or '['?
+            if($offset == 0) {
+                $this->tokens[] = $str{$position};
```

Comment:
Heh, I'm somehow used to that. Dunno where I picked that up :smile: I'll address that right away.